TombRaider1996/ProcessMemory.cs

```diff
@@ -0,0 +1,243 @@
+﻿using System.Diagnostics;
+using System;
+
+namespace TR1
+{
+    /// <summary>
+    ///     The game version's original target platform.
+    /// </summary>
+    internal enum Platform
+    {
+        PC,
+        PSX
+    }
+
+    /// <summary>
+    ///     The supported process versions.
+    /// </summary>
+    internal enum ProcessVersion
+    {
+        ATI,
+        DOSBox,     // Steam's DOSBox
+        ePSXe180,
+        ePSXe190,
+        ePSXe1925,
+        ePSXe200
+    }
+
+    /// <summary>
+    ///     The memory sizes of supported process versions (in bytes).
+    /// </summary>
+    internal enum ExpectedSize
+    {
+        ATI = 3092480,
+        DOSBox = 40321024,
+        ePSXe180 = 10231808,
+        ePSXe190 = 10301440,
+        ePSXe1925 = 10518528,
+        ePSXe200 = 20287488
+    }
+
+    /// <summary>
+    ///     The supported PSX versions of the game.
+    /// </summary>
+    internal enum PSXGameVersion
+    {
+        USA_1_0,
+        USA_final,
+        EU,
+        FR,
+        GER,
+        JP
+    }
+
+    /// <summary>
+    ///     Manages the process's watched memory values for <see cref="Autosplitter"/>'s use.
+    /// </summary>
+    internal class ProcessMemory
+    {
+        private Process _process;
+        public GameData gameData;
+        private EmulatorData _emuData;
+        private ProcessVersion _processVersion;
+        private Platform _platform;
+        private PSXGameVersion? _PSXGameVersion = null;
+        private bool _PSXGameInitialized = false; 
+
+        /// <summary>
+        ///     Updates <see cref="GameData"/> and its addresses' values.
```

Comment:
If fixing the capitalization, ensure this tag's value stays as `GameData` so it refers to the field and not the class (which would be `TR1.GameData`).

Reply:
Done.